ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ServiceGroupResourceProvider.java

```diff
@@ -422,6 +436,11 @@ private void validateCreateRequests(Set<ServiceGroupRequest> requests, Clusters
       }
       serviceGroupNames.get(clusterName).add(serviceGroupName);
 
+      if (request.getMpackNames().size() != 1) {
```

Comment:
Since this change introduces mpacks for service groups, and it also makes exactly one mpack a requirement, it breaks existing code that uses `ServiceGroupResourceProvider` to create mpacks.  For one, it increases the number of failing/erroring unit tests:

```
[ERROR] Tests run: 5122, Failures: 39, Errors: 231, Skipped: 37
->
[ERROR] Tests run: 5123, Failures: 42, Errors: 296, Skipped: 37
```

Can you please address those?  I think most could be fixed by adding some dummy mpack name in https://github.com/apache/ambari/blob/00f29b9a4f7d5829f1c5a45fad6da587f599a224/ambari-server/src/test/java/org/apache/ambari/server/controller/internal/ServiceGroupResourceProviderTest.java#L39-L43